lib/rtcpeerconnection/chrome.js

```diff
@@ -117,12 +124,41 @@ function ChromeRTCPeerConnection(configuration) {
 
 inherits(ChromeRTCPeerConnection, EventTarget);
 
-ChromeRTCPeerConnection.prototype.addStream = function addStream(stream) {
-  if (this._localStreams.has(stream)) {
-    return;
+ChromeRTCPeerConnection.prototype.addTrack = function addTrack() {
+  var args = [].slice.call(arguments);
+  if ('addTrack' in PeerConnection.prototype) {
+    return this._peerConnection.addTrack.apply(this._peerConnection, args);
+  }
+  var mediaStreamTrack = args[0];
+  var syncMediaStreams = args.slice(1);
+
+  if (this._peerConnection.signalingState === 'closed') {
+    throw new Error('Cannot add MediaStreamTrack [' + mediaStreamTrack.id + ', '
+      + mediaStreamTrack.kind + ']: RTCPeerConnection is closed');
   }
-  this._localStreams.add(stream);
-  this._peerConnection.addStream(stream);
+  if (this._tracksToSyncStreams.has(mediaStreamTrack.id)) {
+    throw new Error('Cannot add MediaStreamTrack [' + mediaStreamTrack.id + ', '
+      + mediaStreamTrack.kind + ']: RTCPeerConnection already has it');
+  }
+
+  var syncMediaStreamIds = new Set(syncMediaStreams.map(function(mediaStream) {
+    return mediaStream.id;
+  }));
+  this._tracksToSyncStreams.set(mediaStreamTrack.id, syncMediaStreamIds);
+
+  syncMediaStreamIds.forEach(function(mediaStreamId) {
+    var containerMediaStream = this._syncStreamsToContainerStreams.get(mediaStreamId);
+    if (containerMediaStream) {
+      this._peerConnection.removeStream(containerMediaStream);
+    } else {
+      containerMediaStream = new MediaStream();
+    }
+    containerMediaStream.addTrack(mediaStreamTrack);
+    this._peerConnection.addStream(containerMediaStream);
+    this._syncStreamsToContainerStreams.set(mediaStreamId, containerMediaStream);
+  }, this);
```

Comment:
I don't really understand what's happening here—can you explain? I'm not convinced we need to shim out the full generality of the WebRTC APIs here. We could probably just maintain a single local MediaStream here, IMO. It might simplify the shim implementation. WDYT?

Reply:
@markandrus The basic idea is that since in `addTrack`, the streams passed to it are to be used to sync the particular track, I maintain a MediaStream for each sync MediaStream so that I ensure that all the tracks in that MediaStream are synced.

For ex, if there are calls like this:

```javascript
const sender1 = pc.addTrack(track1, stream1);
const sender2 = pc.addTrack(track2, stream2);
const sender3 = pc.addTrack(track3, stream1);
```

Then `track1` and `track3` must be synchronized. But we can't use `stream1` directly because there is no requirement that `track1` and `track3` be part of `stream1`. So we create a corresponding container `MediaStream`.

Reply:
I believe `addTrack` is supposed to support

```js
addTrack(track, stream1, stream2)
```

as well, but I don't think we can support this particular usage in terms of `addStream` (at least not without SDP munging I think). Given that, and given our limited usage of this API in twilio-video.js, can we go with a simpler shim?

Reply:
@markandrus I think the current implementation does support the case you outlined. It just means that the `track` is supposed to be synced with multiple `streams`, so we create container `streams` for each of the sync `streams` and add the `track` to them.

Reply:
@markandrus In any case, I think I'll go ahead with simplifying the implementation as you suggested, since this is only for until we have `addTrack/removeTrack` support in Chrome available natively.